File: backend/routes/categories.js
// ============================================================
// routes/categories.js — Public category reads
// ============================================================

const express  = require('express');
const mongoose = require('mongoose');
const Category = require('../models/Category');

const router = express.Router();

// GET /api/categories — list all active categories
router.get('/', async (req, res, next) => {
  try {
    if (mongoose.connection.readyState !== 1) {
      return res.status(503).json({ success: false, message: 'Database not connected' });
    }
    const cats = await Category.find({ isActive: true }).sort({ displayOrder: 1, name: 1 });
    res.json({ success: true, data: cats });
  } catch (err) { next(err); }
});

// GET /api/categories/:slug — single category by slug
router.get('/:slug', async (req, res, next) => {
  try {
    if (mongoose.connection.readyState !== 1) {
      return res.status(503).json({ success: false, message: 'Database not connected' });
    }
    const cat = await Category.findOne({ slug: req.params.slug, isActive: true });
    if (!cat) return res.status(404).json({ success: false, message: 'Category not found' });
    res.json({ success: true, data: cat });
  } catch (err) { next(err); }
});

module.exports = router;

File: backend-python/routes/categories.py
from fastapi import APIRouter, HTTPException

from config.database import get_db, is_connected
from utils.helpers import serialize_doc, serialize_list

router = APIRouter()


# GET /api/categories
@router.get("/")
async def list_categories():
    if not is_connected():
        raise HTTPException(503, "Database not connected")

    db = get_db()
    cursor = db.categories.find({"isActive": True}).sort([("displayOrder", 1), ("name", 1)])
    cats = serialize_list(await cursor.to_list(length=200))
    return {"success": True, "data": cats}


# GET /api/categories/:slug
@router.get("/{slug}")
async def get_category(slug: str):
    if not is_connected():
        raise HTTPException(503, "Database not connected")

    db = get_db()
    cat = await db.categories.find_one({"slug": slug, "isActive": True})
    if not cat:
        raise HTTPException(404, "Category not found")
    return {"success": True, "data": serialize_doc(cat)}
